pkgs/development/libraries/boost/generic.nix

```diff
@@ -149,7 +149,7 @@ stdenv.mkDerivation {
   configureScript = "./bootstrap.sh";
   configureFlags = commonConfigureFlags ++ [
     "--with-icu=${icu}"
-    "--with-python=${python}/bin/python"
+    "--with-python=${python}/bin/python${python.majorVersion}"
```

Comment:
boost can be build with python2 and python3. The python2 package has a symlink to python and so works with the previous configuration, however python3 does not have the symlink and so doesn't work.